connectivity/FEATURE_BLE/source/generic/GapImpl.cpp

```diff
@@ -461,23 +463,21 @@ ble_error_t Gap::stopScan()
 {
     ble_error_t err;
 
-    if (!_scan_enabled) {
+    if ((!_scan_enabled && !_scan_pending) || _scan_pending) {
```

Comment:
so what happens when scan is pending and we cancel it and it returns error none but it actually didn't cancel anything?

Reply:
If scan is pending then it can't be canceled by the user. The callback `on_scan_stopped` always reset the `_scan_pending` flag whether the stop operation as completed or not.  
The `_scan_enabled` flag, it is reset when `on_scan_stopped` is called regardless of the operation success. I though of putting an error as we can't retrieve the scanning status from the controller but ended up doing the same choice as the cordio stack. 
What do you think ?

Reply:
IMHO: if scan pending return invalid state

Reply:
or some other error to let user know this call is not available at this time

Reply:
I will return `BLE_STACK_BUSY`